crates/next-api/src/server_actions.rs

```diff
@@ -203,10 +203,14 @@ async fn build_manifest(
 /// The ActionBrowser layer's module is in the Client context, and we need to
 /// bring it into the RSC context.
 pub async fn to_rsc_context(
-    module: Vc<Box<dyn Module>>,
+    client_module: Vc<Box<dyn Module>>,
+    entry_path: &str,
     asset_context: Vc<Box<dyn AssetContext>>,
 ) -> Result<ResolvedVc<Box<dyn Module>>> {
-    let source = FileSource::new_with_query(module.ident().path(), module.ident().query());
+    // TODO a cleaner solution would something similar to the EcmascriptClientReferenceModule, as
+    // opposed to the following hack to construct the RSC module corresponding to this client
+    // module.
+    let source = FileSource::new(client_module.ident().path().root().join(entry_path.into()));
```

Comment:
Is it intentional that we are no longer using the querystring of the original module?

Reply:
No. I added the query string, but had to do this which really sounds like a recipe for duplicated FileSources:
```
FileSource::new_with_query(
    client_module.ident().path().root().join(entry_path.into()),
    Vc::cell(entry_query.into()), // <-- cell!
)
```